pallets/creditcoin/src/ocw.rs

```diff
@@ -167,6 +168,20 @@ fn validate_ethless_transfer(
 		)
 	);
 
+	let nonce = match inputs.get(3) {
+		Some(Token::Uint(value)) => ExternalAmount::from(value),
+		_ => {
+			return Err(OffchainError::InvalidTransfer(
+				"fourth input to ethless transfer was not a Uint",
+			))
```

Comment:
It looks like a unit test asserting on this error condition is missing.

Reply:
I can't actually craft an input that triggers this, it should basically be unreachable. We _could_ panic here instead of erroring since it should be impossible, but erroring is safer in the case that this is ever triggered.